### ISSUE TITLE
[Question}: High CPU But low GPU

### ISSUE THREAD

**Issue by OlivesHere** [run-llama/llama_index#13546]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I just set up my RAG using mistral 7b instruct and llama-index framework in my local PC, and when I am prompting, my CPU usage become 100% but my GPU is just using 0.2GB out of 6GB, but before that i have checked my program is running through cuda not cpu.
And my response time is really long about 50 seconds-2 mins, but in colab i using T4 GPU and it just take within 10 seconds for the same datasets. And below that is my prompt to import my model

llm = LlamaCPP(
    # You can pass in the URL to a GGML model to download it automatically
    #model_url='https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF/resolve/main/mistral-7b-instruct-v0.1.Q4_K_M.gguf',
    #model_url='https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.2-GGUF/resolve/main/mistral-7b-instruct-v0.2.Q4_K_M.gguf',
    # optionally, you can set the path to a pre-downloaded model instead of model_url
    model_path="C:/Users/ASUS608/AppData/Local/llama_index/models/mistral-7b-instruct-v0.1.Q4_K_M.gguf",
    temperature=0.1,
    max_new_tokens=512,
    # llama2 has a context window of 4096 tokens, but we set it lower to allow for some wiggle room
    context_window=4096,
    # kwargs to pass to __call__()
    generate_kwargs={},
    # kwargs to pass to __init__()
    # set to at least 1 to use GPU
    model_kwargs={"n_gpu_layers": 3000},
    # transform inputs into Llama2 format
    messages_to_prompt=messages_to_prompt,
    completion_to_prompt=completion_to_prompt,
    verbose=True,
)
With the program timing

llama_print_timings:        load time =   17151.25 ms
llama_print_timings:      sample time =      33.81 ms /   214 runs   (    0.16 ms per token,  6329.86 tokens per second)
llama_print_timings: prompt eval time =   28892.51 ms /   786 tokens (   36.76 ms per token,    27.20 tokens per second)
llama_print_timings:        eval time =   47526.40 ms /   213 runs   (  223.13 ms per token,     4.48 tokens per second)
llama_print_timings:       total time =   76907.15 ms /   999 tokens

Total runtime of the program is 77.8382477760315s

**Comment by logan-markewich** [comment]
Llama cpp is a little complicated to install. Did you follow the gpu installation instructions?

Imo I would use ollama -- dead simple to install. And don't need to worry about prompt formatting

**Comment by OlivesHere** [comment]
> Llama cpp is a little complicated to install. Did you follow the gpu installation instructions?
> 
> Imo I would use ollama -- dead simple to install. And don't need to worry about prompt formatting

Hi, yes I am following the gpu installation to install, if the following code is no wrong
set CMAKE_ARGS=-DLLAMA_CUBLAS=on

set FORCE_CMAKE=1

pip install llama-cpp-python

**Comment by logan-markewich** [comment]
If you already had llama-cpp installed before that, you'll also need `pip install --no-cache-dir --force-reinstall llama-cpp-python`

**Comment by logan-markewich** [comment]
Seems like their readme has slightly different instructions for windows?
```
`# Windows
$env:CMAKE_ARGS = "-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS"
```

https://github.com/abetlen/llama-cpp-python/blob/main/README.md#installation-configuration

**Comment by OlivesHere** [comment]
> set CMAKE_ARGS=-DLLAMA_CUBLAS=on

let me try again, but really thanks for your help!!!

**Comment by OlivesHere** [comment]
> Seems like their readme has slightly different instructions for windows?
> 
> ```
> `# Windows
> $env:CMAKE_ARGS = "-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS"
> ```
> 
> https://github.com/abetlen/llama-cpp-python/blob/main/README.md#installation-configuration

opps rely wrongly, but thankssssss!!!

**Comment by OlivesHere** [comment]
> If you already had llama-cpp installed before that, you'll also need `pip install --no-cache-dir --force-reinstall llama-cpp-python`

Thanks for your help , using this I am able to use GPU instead of CPU
